### PR TITLE
feat(experience): Added entity text component

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
The entity text component can be used to print any text to the screen, using a configurable entity and field. The text can contain raw text or rich text. 

closes: [HRZ-90347](https://spryker.atlassian.net/browse/HRZ-90347)

[HRZ-90347]: https://spryker.atlassian.net/browse/HRZ-90347?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/platform/experience/entity-image/entity-image.component.ts [hunk 1]
```diff
@@ -12,18 +12,15 @@ export class EntityImageComponent extends ContentMixin<EntityImageOptions>(
...
-  protected $entityField: Signal<string | undefined> = computed<
-    string | undefined
-  >(() => {
-    return this.entityService.getField<string>({
-      element: this,
-      type: this.$options().entity,
-      field: this.$options().field,
-    });
-  });
+  protected $data: Signal<string | undefined> = computed<string | undefined>(
```
Comment (Sinucid) [inline libs/platform/experience/entity-image/entity-image.component.ts]:
```suggestion
  protected $data = computed<string | undefined>(
```

File: libs/platform/experience/entity-text/entity-text.component.spec.ts [hunk 2]
```diff
@@ -1,77 +1,138 @@
...
+      });
+
+      it('should render the prefix inside the heading', () => {
+        const heading = element.shadowRoot?.querySelector('oryx-heading');
+        expect(heading?.textContent?.trim()).toEqual('prefix-foo bar');
+      });
     });
   });
+
+  // describe('when a link option is configured', () => {
```
Comment (Sinucid) [inline libs/platform/experience/entity-text/entity-text.component.spec.ts:124]: Left over?

File: libs/platform/experience/entity-text/entity-text.component.ts [hunk 3]
```diff
@@ -1,28 +1,87 @@
...
+  protected contextService = resolve(ContextService);
 
-  protected $entityField: Signal<string | undefined> = computed<
-    string | undefined
-  >(() => {
-    return this.entityService.getField<string>({
-      element: this,
-      type: this.$options().entity,
-      field: this.$options().field,
+  protected $data: Signal<string | undefined> = computed<string | undefined>(
```
Comment (Sinucid) [inline libs/platform/experience/entity-text/entity-text.component.ts]:
```suggestion
  protected $data = computed<string | undefined>(
```

File: libs/platform/experience/entity-text/entity-text.component.ts [hunk 4]
```diff
@@ -1,28 +1,87 @@
...
   protected override render(): TemplateResult | void {
-    return html`${this.$entityField()}`;
+    const content = this.renderContent();
+    if (!content) return;
+    const { tag } = this.$options();
+    if (tag) {
+      return html`<oryx-heading .tag=${tag}>${content}</oryx-heading>`;
+    } else {
+      return content;
+    }
```
Comment (Sinucid) [inline libs/platform/experience/entity-text/entity-text.component.ts]:
```suggestion
    const content = this.renderContent();
    const { tag } = this.$options();
    return tag && content ? html`<oryx-heading .tag=${tag}>${content}</oryx-heading>`: content;
```

File: libs/platform/experience/entity-text/entity-text.component.ts [hunk 5]
```diff
@@ -1,28 +1,87 @@
...
+  }
+
+  protected renderContent(): TemplateResult | void {
+    const text = this.$text();
+    if (!text) return;
+    // const { link } = this.$options();
```
Comment (Sinucid) [inline libs/platform/experience/entity-text/entity-text.component.ts:54]: Left over?

File: libs/platform/experience/entity-text/entity-text.component.ts [hunk 6]
```diff
@@ -1,28 +1,87 @@
...
+    // const content = html`${this.renderPrefix()}${link
+    //   ? this.renderLink()
+    //   : this.$text()} `;
+
+    return html`${this.renderPrefix()}${this.$text()}`;
+  }
+
+  protected renderLink():
+    | TemplateResult
+    | DirectiveResult<typeof UnsafeHTMLDirective>
```
Comment (Sinucid) [inline libs/platform/experience/entity-text/entity-text.component.ts:64]: `DirectiveResult<typeof UnsafeHTMLDirective>`, what condition returns this ?

File: libs/platform/experience/entity-text/entity-text.component.ts [hunk 7]
```diff
@@ -1,28 +1,87 @@
...
+    return html` <oryx-link><a href=${this.$link()}>${text}</a> </oryx-link> `;
+  }
+
+  protected renderPrefix(): TemplateResult | void {
+    const { prefix } = this.$options();
+    if (prefix) {
+      return html`<span part="prefix" styling="padding-inline-end: 8px;"
+        >${this.$options()?.prefix}</span
+      >`;
+    }
```
Comment (Sinucid) [inline libs/platform/experience/entity-text/entity-text.component.ts]:
```suggestion
    const { prefix } = this.$options();
    if (!prefix) return;
    
    return html`<span part="prefix" styling="padding-inline-end: 8px;">${prefix}</span>`;
```

File: libs/platform/experience/entity-text/entity-text.model.ts [hunk 8]
```diff
@@ -1,8 +1,8 @@
...
-}
+import { HeadingTag } from '@spryker-oryx/ui/heading';
+import { EntityFieldOptions } from '../src/models';
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface EntityTextOptions extends EntityFieldOptions {}
+export interface EntityTextOptions extends EntityFieldOptions {
+  tag?: HeadingTag;
+  prefix?: string;
+  // link?: boolean;
```
Comment (Sinucid) [inline libs/platform/experience/entity-text/entity-text.model.ts:7]: ?

File: libs/platform/experience/entity-text/entity-text.schema.ts [hunk 9]
```diff
@@ -14,5 +15,27 @@ export const entityTextSchema: ContentComponentSchema<EntityTextComponent> = {
...
+        { value: HeadingTag.Strong },
+        { value: HeadingTag.Subtitle },
+        { value: HeadingTag.SubtitleSmall },
+        { value: HeadingTag.Caption },
+      ],
+    },
+    prefix: {
+      type: FormFieldType.Text,
+    },
+    // link: {
```
Comment (Sinucid) [inline libs/platform/experience/entity-text/entity-text.schema.ts:37]: ?

File: libs/platform/experience/entity-text/stories/static.stories.ts [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+import { Story } from '@storybook/web-components';
+import { TemplateResult, html } from 'lit';
+import { storybookPrefix } from '../../.constants';
+import { EntityTextOptions } from '../entity-text.model';
+
+export default {
+  title: `${storybookPrefix}/Entity Text`,
+};
+
+type Props = EntityTextOptions /*& FieldComponentProperties*/;
```
Comment (Sinucid) [inline libs/platform/experience/entity-text/stories/static.stories.ts]: ?

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/platform/experience/entity-text/entity-text.component.spec.ts (hunk 2 above)
Comment (tobi-or-not-tobi) [reply libs/platform/experience/entity-text/entity-text.component.spec.ts:124]: yeah, we'll add link soon, hence we kept a few left overs for that

File: libs/platform/experience/entity-text/entity-text.component.ts (hunk 6 above)
Comment (tobi-or-not-tobi) [reply libs/platform/experience/entity-text/entity-text.component.ts:64]: `this.$text();`

**Review by Sinucid — DISMISSED** [review]

**Review by dunqan — APPROVED** [review]

**Pull request merged**